src/main/java/com/tdep/tadlab/dao/ProjectDao.java

```diff
@@ -0,0 +1,16 @@
+package com.tdep.tadlab.dao;
+
+import com.tdep.tadlab.model.Project;
+import com.tdep.tadlab.model.Tool;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface ProjectDao {
```

Comment:
nit: based on vague memories from many years ago + brief skimming on google, if you're planning on leveraging Spring Data JPA, this should be something more like `interface ProjectRepository extends JpaRepository` 

Basically, if this class is intended to manage the actual DB queries, seems like the naming pattern is `repository` -- there's different solutions out of the box depending on how granular or basic your queries need to be, so you might want to look into that a bit.

Reply:
I think you're right - that is a pattern I've been coming across as I look things up. I think I was trying to originally use a DAO structure but sort of stumbled into Spring Boot and found a way to sort of use both? I'll try this with the Project service and see if I'll extend it to the existing Job service.

Reply:
I think the DAO pattern is also legit, but maybe less favored than the repository approach (me spitballing based off the fact that the official Spring tutorials seem to all reference the repository pattern). I think Spring tried to make the basic CRUD operations as out-of-the-box as possible so the repository approach gives you a lot of free queries right off the bat

Reply:
Agreed, I've set up the new service to follow the repository pattern and will create the new tables to match when I merge this pr. Fingers crossed!